core/src/main/java/io/aiven/klaw/service/AclSyncControllerService.java

```diff
@@ -216,6 +216,31 @@ private void approveSyncBackAcls(
       } else if (resultAclNullCheck.contains("Acl already exists")) {
         logUpdateSyncBackTopics.add(String.format(ACL_SYNC_ERR_103, aclFound.getTopicname()));
       } else {
+        Env env =
+            manageDatabase
+                .getHandleDbRequests()
+                .getEnvDetails(syncBackAcls.getSourceEnv(), tenantId);
+        KwClusters kwClusters =
+            manageDatabase
+                .getClusters(KafkaClustersType.of(env.getType()), tenantId)
+                .get(env.getClusterId());
+        // Update aivenaclid in klaw metadata
+        if (kwClusters.getKafkaFlavor().equals(KafkaFlavors.AIVEN_FOR_APACHE_KAFKA.value)) {
+          String jsonParams = "{}", aivenAclIdKey = "aivenaclid";
+          if (Objects.requireNonNull(responseBody).isSuccess()) {
+            Object responseData = responseBody.getData();
+            if (responseData instanceof Map) {
+              Map<String, String> dataMap = (Map<String, String>) responseData;
+              if (dataMap.containsKey(aivenAclIdKey)) {
+                jsonParams = "{\"" + aivenAclIdKey + "\":\"" + dataMap.get(aivenAclIdKey) + "\"}";
```

Comment:
Would we be better creating a POJO for this and converting it using an object mapper?

Reply:
Updated